src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -124,6 +125,9 @@ public override void ExecuteCmdlet()
             newEnvironment.Endpoints[AzureEnvironment.Endpoint.ManagementPortalUrl] = ManagementPortalUrl;
             newEnvironment.Endpoints[AzureEnvironment.Endpoint.StorageEndpointSuffix] = StorageEndpoint;
             newEnvironment.Endpoints[AzureEnvironment.Endpoint.ActiveDirectory] = ActiveDirectoryEndpoint;
+            newEnvironment.Endpoints[AzureEnvironment.Endpoint.ActiveDirectory] =
+                ActiveDirectoryEndpoint != null ? GeneralUtilities.EnsureTrailingSlash(ActiveDirectoryEndpoint)
```

Comment:
@cormacpayne let's take this opportunity to fix GeneralUtilities.EnsureTrailingSlash.
We need a null check inside the function for url. if we do that, we do not have to null check when call the function.